File: app/src/main/java/com/example/pam_project/networking/authors/RestAuthorsRepository.java
package com.example.pam_project.networking.authors;

import com.example.pam_project.networking.APIService;
import com.example.pam_project.networking.RetrofitUtils;

import java.util.List;

import io.reactivex.Single;

public class RestAuthorsRepository implements AuthorsRepository {
    private final APIService service;
    private final AuthorsMapper authorsMapper;

    public RestAuthorsRepository(final APIService service, final AuthorsMapper configMapper) {
        this.service = service;
        this.authorsMapper = configMapper;
    }

    @Override
    public Single<List<AuthorsModel>> getAuthors() {
        return RetrofitUtils.performRequest(service.getAuthors()).map(authorsMapper::map);
    }
}

File: app/src/main/java/com/example/pam_project/networking/authors/RestAuthorsRepository.kt
package com.example.pam_project.networking.authors

import com.example.pam_project.networking.APIService
import com.example.pam_project.networking.RetrofitUtils
import io.reactivex.Single

class RestAuthorsRepository(private val service: APIService?, private val authorsMapper: AuthorsMapper?) : AuthorsRepository {
    override val authors: Single<List<AuthorsModel?>?>
        get() = RetrofitUtils.performRequest(service.getAuthors()).map { response: ListAuthorsResponse? -> authorsMapper!!.map(response) }
}